backend/middleware/retry-example.ts

```
@@ -0,0 +1,225 @@
+    baseDelay: 2000,
+    timeout: 30000,
+    shouldRetry: (error) => {
+      // Retry on rate limits and server errors
+      return error.status === 429 || (error.status >= 500 && error.status < 600);
+    }
+  });
+
+  const mockGPT4oCall = async () => {
+    // Simulate GPT-4o API call
+    if (Math.random() < 0.6) {
+      const error: any = new Error('Rate limit exceeded');
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)